drv/auxflash-server/src/main.rs

```diff
@@ -254,7 +258,21 @@ impl idl::InOrderAuxFlashImpl for ServerImpl {
     ) -> Result<AuxFlashId, RequestError<AuxFlashError>> {
         let mut idbuf = [0; 20];
         self.qspi.read_id(&mut idbuf).map_err(qspi_to_auxflash)?;
-        Ok(AuxFlashId(idbuf))
+        let mfr_id = idbuf[0];
+        let memory_type = idbuf[1];
+        let capacity = idbuf[2];
+
+        let mut unique_id = [0; 12];
+        self.qspi
+            .read_unique_id(&mut unique_id)
+            .map_err(qspi_to_auxflash)?;
+        let unique_id: [u8; 8] = unique_id[4..].try_into().unwrap_lite();
```

Comment:
so, it occurs to me that we could save 12 bytes of stack if we changed this to reuse `idbuf`, like
```suggestion
        let mut unique_id: &mut [u8; 12] = &mut idbuf[..12].try_into().unwrap_lite();
        self.qspi
            .read_unique_id(unique_id)
            .map_err(qspi_to_auxflash)?;
        let unique_id: [u8; 8] = unique_id[4..].try_into().unwrap_lite();
```
this comes at the cost of an additional panic site, though; idk if it's worth it... (it's probably not)

Reply:
I think this isn't worth it; the stack for this function is already huge (due to a Big Variant and lack of placement new).